### PR TITLE
Changes in docs [Form Recognizer]

### PR DESCRIPTION
**Pull request by rakshith91**

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 1]
```diff
@@ -145,6 +145,13 @@ The client exposes a `begin_<method-name>` method that returns an `LROPoller` or
 for the operation to complete by calling `result()` on the operation returned from the `begin_<method-name>` method. 
 Sample code snippets are provided to illustrate using long-running operations [below](#examples "Examples").
 
+### Async APIs
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:148]: It doesn't feel right to have this under `Key Concepts`... Maybe at the end of Examples? Since we have a bunch of sync examples on the README then this could help point to the async ones

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 2]
```diff
@@ -184,6 +191,15 @@ for recognized_form in result:
         ))
 ```
 
+Alternatively, a form url can also be used to recognize custom forms using the `begin_recognize_custom_forms_from_url` method.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:194]: Can we add a quick sentence about the `from_url` existing for all recognize methods?

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 3]
```diff
@@ -244,6 +260,9 @@ Train a machine-learned model on your own form type. The resulting model will be
 Provide a container SAS url to your Azure Storage Blob container where you're storing the training documents. See details on setting this up
 in the [service quickstart documentation][quickstart_training].
 
+To train a model, you will need Azure Blob storage container's shared access signature (SAS) URL. See [here][train-a-model-using-labeled-data] for more detailed descriptions on how to get it.
+Alternatively, a container uri can be used instead if the container is public.
+
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:265]: I suggest rebasing on master, I edited this part of the README in a different PR and I'm not sure my edit + your edit make sense now 😆

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py [hunk 4]
```diff
@@ -93,11 +93,13 @@ def __init__(self, endpoint, credential, **kwargs):
...
         """Create and train a custom model. The request must include a `training_files_url` parameter that is an
-        externally accessible Azure storage blob container Uri (preferably a Shared Access Signature Uri).
+        externally accessible Azure storage blob container Uri (preferably a Shared Access Signature Uri). Note that
+        a container uri is accepted only when the container is public.
         Models are trained using documents that are of the following content type - 'application/pdf',
         'image/jpeg', 'image/png', 'image/tiff'. Other type of content in the container is ignored.
 
-        :param str training_files_url: An Azure Storage blob container's SAS URI.
+        :param str training_files_url: An Azure Storage blob container's SAS URI. A container uri can be used if the
+            container is public.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:102]: 👍

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py [hunk 5]
```diff
@@ -278,7 +280,7 @@ def get_copy_authorization(self, resource_id, resource_region, **kwargs):
         :param str resource_id: Azure Resource Id of the target Form Recognizer resource
             where the model will be copied to.
         :param str resource_region: Location of the target Form Recognizer resource. A valid Azure
-            region name supported by Cognitive Services.
+            region name supported by Cognitive Services. For example, 'westus', 'centraluseuap' etc.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:283]: I don't want to suggest people create FR resources in canary. Let's change 'centraluseuap' to like 'eastus' or something

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 6]
```diff
@@ -242,7 +242,9 @@ class FieldText(FormContent):
         Units are in pixels for images and inches for PDF.
     :ivar text_content:
         When `include_text_content` is set to true, a list of text
-        elements constituting this field or value is returned.
+        elements constituting this field or value is returned. The list
+        constitutes of text elements such as lines and words in addition
+        to form fields.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:247]: we shouldn't be getting FormFields back in text_content. what do you mean here?

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py (hunk 5 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:283]: Let's also add this link that shows regional availability: https://azure.microsoft.com/global-infrastructure/services/?products=cognitive-services

Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:283]:
```suggestion
            region name supported by Cognitive Services. For example, 'westus', 'eastus' etc. See https://azure.microsoft.com/en-us/global-infrastructure/services/?products=cognitive-services for the regional availability of Cognitive Services
```

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py (hunk 6 above)
Comment (rakshith91) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:247]: let me rephrase it

Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:247]:
```suggestion
        elements constituting this field or value is returned. The list
        constitutes of text elements such as lines and words.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 7]
```diff
@@ -464,7 +466,9 @@ class FormTableCell(FormContent):
         The 1-based number of the page in which this content is present.
     :ivar text_content:
         When `include_text_content` is set to true, a list of text
-        elements constituting this cell is returned.
+        elements constituting this cell is returned. The list
+        constitutes of text elements such as lines and words in addition
+        to form fields.
```
Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:471]:
```suggestion
        elements constituting this cell is returned. The list
        constitutes of text elements such as lines and words.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md (hunk 3 above)
Comment (rakshith91) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:261]: weird - it doesn't show any conflicts even when i rebase it - ill just remove it for now and fix it later

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 8]
```diff
@@ -246,6 +256,9 @@ If training files are within a subfolder in the container, use the [prefix][pref
 
 More details on setting up a container and required file structure can be found in the [service quickstart documentation][quickstart_training].
 
+To train a model, you will need Azure Blob storage container's shared access signature (SAS) URL. See [here][train-a-model-using-labeled-data] for more detailed descriptions on how to get it.
+Alternatively, a container uri can be used instead if the container is public.
+
```
Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:261]:
```suggestion
```

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py [hunk 9]
```diff
@@ -278,7 +280,7 @@ def get_copy_authorization(self, resource_id, resource_region, **kwargs):
         :param str resource_id: Azure Resource Id of the target Form Recognizer resource
             where the model will be copied to.
         :param str resource_region: Location of the target Form Recognizer resource. A valid Azure
-            region name supported by Cognitive Services.
+            region name supported by Cognitive Services. For example, 'westus', 'eastus' etc. See https://azure.microsoft.com/en-us/global-infrastructure/services/?products=cognitive-services for the regional availability of Cognitive Services
```
Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:283]:
```suggestion
            region name supported by Cognitive Services. For example, 'westus', 'eastus' etc. 
            See https://azure.microsoft.com/en-us/global-infrastructure/services/?products=cognitive-services
            for the regional availability of Cognitive Services
```

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py [hunk 10]
```diff
@@ -278,7 +280,9 @@ def get_copy_authorization(self, resource_id, resource_region, **kwargs):
         :param str resource_id: Azure Resource Id of the target Form Recognizer resource
             where the model will be copied to.
         :param str resource_region: Location of the target Form Recognizer resource. A valid Azure
-            region name supported by Cognitive Services.
+            region name supported by Cognitive Services. For example, 'westus', 'eastus' etc.
```
Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:283]:
```suggestion
            region name supported by Cognitive Services. For example, 'westus', 'eastus' etc.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 11]
```diff
@@ -407,6 +425,7 @@ This project has adopted the [Microsoft Open Source Code of Conduct][code_of_con
 [python-fr-product-docs]: https://docs.microsoft.com/azure/cognitive-services/form-recognizer/overview
 [python-fr-ref-docs]: https://aka.ms/azsdk/python/formrecognizer/docs
 [python-fr-samples]: https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/formrecognizer/azure-ai-formrecognizer/samples
+[train-a-model-using-labeled-data]: https://docs.microsoft.com/azure/cognitive-services/form-recognizer/quickstarts/python-labeled-data#train-a-model-using-labeled-data
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:428]: doesn't look like this is used

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py [hunk 12]
```diff
@@ -278,7 +280,9 @@ def get_copy_authorization(self, resource_id, resource_region, **kwargs):
...
+            See https://azure.microsoft.com/en-us/global-infrastructure/services/?products=cognitive-services
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:284]: nit: remove the 'en-us' in the url

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py [hunk 13]
```diff
@@ -300,7 +302,7 @@ async def get_copy_authorization(
         :param str resource_id: Azure Resource Id of the target Form Recognizer resource
             where the model will be copied to.
         :param str resource_region: Location of the target Form Recognizer resource. A valid Azure
-            region name supported by Cognitive Services.
+            region name supported by Cognitive Services. For example, 'westus', 'centraluseuap' etc.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py:305]: add the link for async too?
